common/src/main/java/io/netty/util/TrackedIllegalReferenceCountException.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util;
+
+import io.netty.util.internal.StringUtil;
+
+/**
+ * A special {@link IllegalReferenceCountException} with the ability to track access records
+ */
+public class TrackedIllegalReferenceCountException extends IllegalReferenceCountException {
```

Comment:
should we make this package-private for now and final ?

Reply:
Since this exception will be used in `AdvancedLeakAwareByteBuf`, this can't be a package-private class..